### PR TITLE
Fix issues with phone number not being extracted (#2068)

### PR DESCRIPTION
**Pull request by visionshao**
Fix the issue #2068 
Change the regexes of NLPhoneNumber and ITPhoneNumber;
Add codes for checking if extracted string has a non-digit string before "-" ;
Add Specs to Specs/Sequence/English/PhoneNumberModel.json;

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.Sequence/Extractors/BasePhoneNumberExtractor.cs [hunk 1]
```diff
@@ -130,8 +130,11 @@ public override List<ExtractResult> Extract(string text)
                             var charGap = text[(int)(er.Start - 2)];
                             if (!char.IsNumber(charGap) && !char.IsWhiteSpace(charGap))
                             {
+                                // check if the extracted string has a non-digit string before "-".
+                                var flag = Regex.IsMatch(text.Substring(0, (int)(er.Start - 2)), @"^[^0-9]+$");
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.Sequence/Extractors/BasePhoneNumberExtractor.cs:134]: Checking some kind of prefix for the tagged potential phonenumber only if fortmatted? Is that it?

File: Specs/Sequence/English/PhoneNumberModel.json [hunk 2]
```diff
@@ -4776,5 +4776,42 @@
   {
     "Input": "My card number is 12345678",
     "Results": []
+  },
+  {
+    "Input": "OFFICE-1(516)733-3989",
```
Comment (tellarin) [inline Specs/Sequence/English/PhoneNumberModel.json:4781]: Can you add a few more cases to show that your changes handle more general cases?
Ex: "OFFICE - 1(516)733-3989" or "Office:(516)733-3989"

File: .NET/Microsoft.Recognizers.Text.Sequence/Extractors/BasePhoneNumberExtractor.cs (hunk 1 above)
Comment (visionshao) [reply .NET/Microsoft.Recognizers.Text.Sequence/Extractors/BasePhoneNumberExtractor.cs:134]: yes

File: Specs/Sequence/English/PhoneNumberModel.json (hunk 2 above)
Comment (visionshao) [reply Specs/Sequence/English/PhoneNumberModel.json:4781]: ok

**Pull request merged**